### PR TITLE
Fix and expand user data path information

### LINKED ISSUE #5462: Data Paths documentation page for Windows is inconsistent with ProjectSettings and OS dosumentation
**Your Godot version:** 3.4

**Issue description:** documentation inconsistency

**URL to the documentation page:**  https://docs.godotengine.org/en/stable/tutorials/io/data_paths.html

The page says:

> On desktop operating systems, the engine uses the typical ~/.local/share/godot/app_userdata/Name on macOS and Linux, and %APPDATA%/Name on Windows. Name is taken from the application name defined in the Project Settings, but it can be overridden on a per-platform basis using feature tags.

and:
> User Data - Windows: %APPDATA%\Godot\

While the real behavior is `%APPDATA%\Godot\app_userdata\[project_name]`, or `%APPDATA%\[custom_name]` if `use_custom_user_dir` is set, as correctly described here: https://docs.godotengine.org/en/stable/classes/class_os.html#class-os-method-get-user-data-dir

Note that it is also correctly described here: https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-property-application-config-custom-user-dir-name

and here: https://docs.godotengine.org/en/3.1/classes/class_projectsettings.html#class-projectsettings-property-application-config-custom-user-dir-name

### PR DESCRIPTION
**Pull request by skyace65**
Fixes the user data location for windows. Also adds information on the data path when the `use_custom_user_dir` project setting is true. Closes #5462

### CODE REVIEW

File: tutorials/io/data_paths.rst [hunk 1]
```diff
@@ -38,9 +38,10 @@ when the game is running, the project's file system will likely be read-only.
 The ``user://`` prefix points to a different directory on the user's device. On
 mobile and consoles, this path is unique to the project. On desktop, the engine
 stores user files in ``~/.local/share/godot/app_userdata/Name`` on
-Linux, ``~/Library/Application Support/Godot/app_userdata/Name`` on macOS (since Catalina) and ``%APPDATA%/Name`` on Windows. ``Name`` is based on the application
-name defined in the Project Settings, but you can override it on a per-platform
-basis using :ref:`feature tags <doc_feature_tags>`.
+Linux, ``~/Library/Application Support/Godot/app_userdata/Name`` on macOS
+(since Catalina) and ``%APPDATA%\Godot\app_userdata\[project_name]`` on Windows.
+``Name`` is based on the application name defined in the Project Settings, but
```
Comment (Exerionius) [inline tutorials/io/data_paths.rst:43]: Shouldn't it be `project_name` instead of `Name`?

File: tutorials/io/data_paths.rst [hunk 2]
```diff
@@ -65,21 +66,25 @@ Editor data paths
...
-|               | - Linux: ``~/.cache/godot/``                      |
-+---------------+---------------------------------------------------+
++------------------------------+-----------------------------------------------------------+
+| Type                         | Location                                                  |
++==============================+===========================================================+
+| User data                    | - Windows: ``%APPDATA%\Godot\app_userdata\``              |
+|                              | - macOS: ``~/Library/Application Support/Godot/``         |
+|                              | - Linux: ``~/.local/share/godot/``                        |
++------------------------------+-----------------------------------------------------------+
+| User Data                    | - Windows: ``%APPDATA%\``                                 |
```
Comment (Exerionius) [inline tutorials/io/data_paths.rst:76]: `%APPDATA%\[custom_name]` or `%APPDATA%\[custom_dir]` would be a bit clearer, don't you think?

**Comment by skyace65** [comment]
Fixed. Both parts of that page now use `[project_name]`

File: tutorials/io/data_paths.rst [hunk 3]
```diff
@@ -65,21 +66,25 @@ Editor data paths
...
-|               | - Linux: ``~/.cache/godot/``                      |
-+---------------+---------------------------------------------------+
++------------------------------+----------------------------------------------------------------+
+| Type                         | Location                                                       |
++==============================+================================================================+
+| User data                    | - Windows: ``%APPDATA%\Godot\app_userdata\[project_name]``     |
+|                              | - macOS: ``~/Library/Application Support/Godot/[project_name]``|
+|                              | - Linux: ``~/.local/share/godot/[project_name]``               |
++------------------------------+----------------------------------------------------------------+
+| User Data                    | - Windows: ``%APPDATA%\[project_name]``                        |
```
Comment (Calinou) [inline tutorials/io/data_paths.rst:76]:
```suggestion
| User data                    | - Windows: ``%APPDATA%\[project_name]``                        |
```

File: tutorials/io/data_paths.rst [hunk 4]
```diff
@@ -65,21 +66,25 @@ Editor data paths
...
+| project setting is true)     | - Linux: ``~/.local/share/[project_name]``                     |
++------------------------------+----------------------------------------------------------------+
+| User settings                | - Windows: ``%APPDATA%\Godot\``                                |
+|                              | - macOS: ``~/Library/Application Support/Godot/``              |
+|                              | - Linux: ``~/.config/godot/``                                  |
++------------------------------+----------------------------------------------------------------+
+| Cache                        | - Windows: ``%TEMP%\Godot\``                                   |
+|                              | - macOS: ``~/Library/Caches/Godot/``                           |
+|                              | - Linux: ``~/.cache/godot/``                                   |
++------------------------------+----------------------------------------------------------------+
```
Comment (Calinou) [inline tutorials/io/data_paths.rst:87]:
```suggestion
+-------------------------------+----------------------------------------------------------------+
| Type                          | Location                                                       |
+===============================+================================================================+
| User data                     | - Windows: ``%APPDATA%\Godot\app_userdata\[project_name]``     |
|                               | - macOS: ``~/Library/Application Support/Godot/[project_name]``|
|                               | - Linux: ``~/.local/share/godot/[project_name]``               |
+-------------------------------+----------------------------------------------------------------+
| User data                     | - Windows: ``%APPDATA%\[project_name]``                        |
| (when ``use_custom_user_dir`` | - macOS: ``~/Library/Application Support/[project_name]``      |
| project setting is ``true``)  | - Linux: ``~/.local/share/[project_name]``                     |
+-------------------------------+----------------------------------------------------------------+
| User settings                 | - Windows: ``%APPDATA%\Godot\``                                |
|                               | - macOS: ``~/Library/Application Support/Godot/``              |
|                               | - Linux: ``~/.config/godot/``                                  |
+-------------------------------+----------------------------------------------------------------+
| Cache                         | - Windows: ``%TEMP%\Godot\``                                   |
|                               | - macOS: ``~/Library/Caches/Godot/``                           |
|                               | - Linux: ``~/.cache/godot/``                                   |
+-------------------------------+----------------------------------------------------------------+
```

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Great, thank you!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Pull request merged**